holmes/plugins/toolsets/bash/common/stringify.py

```diff
@@ -0,0 +1,25 @@
+import shlex
+
+SAFE_SHELL_CHARS = frozenset(".-_=/,:")
+
+
+def escape_shell_args(args: list[str]) -> list[str]:
+    """
+    Escape shell arguments to prevent injection.
+    Uses shlex.quote for safe shell argument quoting.
+    """
+    escaped_args = []
+
+    for arg in args:
+        # If argument is safe (contains only alphanumeric, hyphens, dots, underscores, equals, slash, comma, colon)
+        # no escaping needed
+        if arg and all(c.isalnum() or c in SAFE_SHELL_CHARS for c in arg):
```

Comment:
Isn't shlex.quote already doing this logic as well? 
I see that it search for safe character.

Reply:
Yes, it's a bit duplicated.